plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -236,13 +236,39 @@ public void setupLaunchConfiguration(ILaunchConfigurationWorkingCopy workingCopy
     workingCopy.setMappedResources(projects.toArray(new IResource[projects.size()]));
   }
 
+  @VisibleForTesting
+  void checkPorts(DefaultRunConfiguration devServerRunConfiguration, PortChecker portInUse)
+      throws CoreException {
+    InetAddress serverHost = InetAddress.getLoopbackAddress();
+    if (devServerRunConfiguration.getHost() != null) {
+      serverHost = LocalAppEngineServerLaunchConfigurationDelegate
+          .resolveAddress(devServerRunConfiguration.getHost());
+    }
+    serverPort = checkPort(serverHost,
+        ifNull(devServerRunConfiguration.getPort(), DEFAULT_SERVER_PORT), portInUse);
+
+    if (LocalAppEngineServerLaunchConfigurationDelegate.DEV_APPSERVER2) {
```

Comment:
This code snippet was taken from below. The only difference is this `if` statement here. We check `adminPort` only when it is dev_appserver2.